auraescript/macros/src/ops.rs

```diff
@@ -209,7 +220,14 @@ fn typescript_service_generator(
 ) -> String {
     let service_name = service.name();
     let mut ts_funcs: String = format!(
-        "export class {service_name}Client implements {service_name} {{"
+        r#"
+export class {service_name}Client implements {service_name} {{
+    client: u32
+
+    constructor(client: u32) {{
```

Comment:
Should we use constructor overloading like [here](https://stackoverflow.com/a/12702786)?

I'm trying to think of a way of not forcing every script to have to create the client. The op_ident above can be changed to take `client: Option<::deno_core::ResourceId>`, in which case it can fall back to using `::client::Client::default().await?;`.

Is this a good idea?

Reply:
I tried what you described in the last commit, and it seem to work great  :100: